packages/relay/tests/lib/eth.spec.ts

```diff
@@ -258,4 +352,40 @@ describe('Eth', async function () {
       expect(receipt).to.deep.eq(cachedReceipt);
     });
   });
+
+  describe('eth_getTransactionByHash', async function () {
+    it('returns `null` for non-existing hash', async function () {
+      // mirror node request mocks
+      mock.onGet(`contracts/results/${defaultTxHash}`).reply(200, defaultDetailedContractResultByHash);
```

Comment:
Based on your description this should match a 404 with a status error message returned from mirror node
```suggestion
      mock.onGet(`contracts/results/${defaultTxHash}`).reply(404, {
      '_status': {
        'messages': [
          {
            'message': 'No such detailed contract result exists'
          }
        ]
      }
      });
```